presto-tpch/src/test/java/com/facebook/presto/tpch/statistics/MinMaxSet.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tpch.statistics;
+
+import java.util.Optional;
+
+public class MinMaxSet<E>
+{
+    private Optional<E> min = Optional.empty();
+    private Optional<E> max = Optional.empty();
+
+    public MinMaxSet()
+    {
+    }
+
+    public MinMaxSet(Optional<E> min, Optional<E> max)
+    {
+        this.min = min;
+        this.max = max;
+    }
+
+    public Optional<E> getMin()
+    {
+        return min;
+    }
+
+    public Optional<E> getMax()
+    {
+        return max;
+    }
+
+    public void add(E ele)
+    {
+        final Comparable<? super E> castedEle = (Comparable<? super E>) ele;
+        if (!min.isPresent() || castedEle.compareTo(min.get()) <= 0) {
+            min = Optional.of(ele);
+        }
+        if (!max.isPresent() || castedEle.compareTo(max.get()) >= 0) {
+            max = Optional.of(ele);
```

Comment:
Please do not abbreviate variable names.
```suggestion
    public void add(E element)
    {
        final Comparable<? super E> castedElement = (Comparable<? super E>) element;
        if (!min.isPresent() || castedElement.compareTo(min.get()) <= 0) {
            min = Optional.of(element);
        }
        if (!max.isPresent() || castedElement.compareTo(max.get()) >= 0) {
            max = Optional.of(element);
```